pallets/hyperbridge_aggregations/src/benchmarking.rs

```diff
@@ -0,0 +1,159 @@
+    use sp_runtime::traits::Bounded;
+
+    pub static DEFAULT_EMPTY_ATT: [u8; 32] =
+        hex!("290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563");
+
+    pub static TEST_CONTRACT: [u8; 20] = hex!("d9145CCE52D386f254917e481eB44e9943F39138");
+
+    /// Return a whitelisted account with enough founds to do anything.
+    pub fn funded_account<T: Config>() -> T::AccountId {
+        let caller: T::AccountId = whitelisted_caller();
+        T::Currency::set_balance(&caller, BalanceOf::<T>::max_value() / 2u32.into());
+        caller
+    }
+}
+
+#[benchmarks]
+mod benchmarks {
+    use super::utils::*;
+    use super::*;
+    use sp_runtime::traits::Bounded;
+
+    #[benchmark]
+    fn dispatch_aggregation() {
+        let caller: T::AccountId = funded_account::<T>();
+
+        // Prepare the parameters
+        let params = Params {
+            aggregation_id: 1u64,
+            aggregation: sp_core::H256(DEFAULT_EMPTY_ATT),
+            module: sp_core::H160(TEST_CONTRACT),
+            destination: StateMachine::Kusama(4009),
+            timeout: 0,
+            fee: Zero::zero(),
+        };
+
+        let initial_nonce = pallet_ismp::Nonce::<T>::get();
+        let initial_event_count = frame_system::Pallet::<T>::events().len();
+
+        #[extrinsic_call]
+        dispatch_aggregation(RawOrigin::Signed(caller), params);
+
+        // Assertions
+        assert!(
+            frame_system::Pallet::<T>::events().len() > initial_event_count,
+            "No events were emitted"
+        );
+        assert_eq!(
+            pallet_ismp::Nonce::<T>::get(),
+            initial_nonce + 1,
+            "Nonce should be incremented"
+        );
+    }
+
+    #[benchmark]
+    fn dispatch_aggregation_large_timeout() {
+        let caller: T::AccountId = funded_account::<T>();
+
+        let one_year_seconds = 365 * 24 * 60 * 60;
+        let params = Params {
+            aggregation_id: 1u64,
+            aggregation: sp_core::H256(DEFAULT_EMPTY_ATT),
+            module: sp_core::H160(TEST_CONTRACT),
+            destination: StateMachine::Kusama(4009),
+            timeout: one_year_seconds,
+            fee: Zero::zero(),
+        };
+
+        let initial_nonce = pallet_ismp::Nonce::<T>::get();
+        let initial_event_count = frame_system::Pallet::<T>::events().len();
+
+        #[extrinsic_call]
+        dispatch_aggregation(RawOrigin::Signed(caller.clone()), params.clone());
+
+        // Assertions
+        assert!(
+            frame_system::Pallet::<T>::events().len() > initial_event_count,
+            "No events were emitted"
+        );
+        assert_eq!(
+            pallet_ismp::Nonce::<T>::get(),
+            initial_nonce + 1,
+            "Nonce should be incremented"
+        );
+    }
+
+    #[benchmark]
+    fn dispatch_aggregation_large_fee() {
+        let caller: T::AccountId = funded_account::<T>();
+        let large_fee = BalanceOf::<T>::max_value() / 4u32.into();
```

Comment:
Minor: I don't expect any significant difference in weight depending on the fee amount

Reply:
Based on the results and my previous comment looks like there is some still, which is why I'm keeping the fee benchmark


```
fn dispatch_aggregation() -> Weight {
        // Proof Size summary in bytes:
        //  Measured:  `9`
        //  Estimated: `3474`
        // Minimum execution time: 9_000_000 picoseconds.
        Weight::from_parts(10_000_000, 3474)
            .saturating_add(T::DbWeight::get().reads(2_u64))
            .saturating_add(T::DbWeight::get().writes(2_u64))
    }
    
    fn dispatch_aggregation_large_fee() -> Weight {
        // Proof Size summary in bytes:
        //  Measured:  `9`
        //  Estimated: `3593`
        // Minimum execution time: 31_000_000 picoseconds.
        Weight::from_parts(32_000_000, 3593)
            .saturating_add(T::DbWeight::get().reads(3_u64))
            .saturating_add(T::DbWeight::get().writes(3_u64))
    }
    ```